pints/tests/test_error_measures.py

```diff
@@ -243,6 +261,24 @@ def test_sum_of_squares_error(self):
         self.assertEqual(dy[0], -12)
         self.assertEqual(dy[1], -12)
 
+        # Test weighting
+        e = pints.SumOfSquaresError(p, weights=[1, 2])
+        self.assertRaises(ValueError, pints.SumOfSquaresError, p, [1, 2, 3])
+        # Residuals are: [[0, 0], [-1, -1], [-2, -2]]
+        # Error is (1+4)+(1+4)*2=15
+        self.assertEqual(e(x), 15)
+        y, dy = e.evaluateS1(x)
+        self.assertEqual(y, 15)
+        # derivatives are [[[1, 1], [1, 1]], [[1, 1], [1, 1]], [[1, 1], [1,1]]]
+        # dy1 is: 2 * [(0*1+-1*1+-2*1)*1 + (0*1+-1*1+-2*1)*2]
+        #       = 2 * [-3*1 + -3*2]
+        #       = -18
+        # dy1 is: 2 * [(0*1+-1*1+-2*1)*1 + (0*1+-1*1+-2*1)*2]
+        #       = 2 * [-3*1 + -3*2]
+        #       = -18
+        self.assertEqual(dy[0], -18)
+        self.assertEqual(dy[1], -18)
```

Comment:
Shouldn't these two be different because of the weight?

Reply:
Just for the record, they are the same because of the constant model. The weights and values will be summed in the same way in the two parameters, and the only thing that matters here is the derivatives of the parameters -- which are the same here (constant model), so no difference.